src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -791,15 +791,41 @@ def _validate_lock_params_match_lock(
                 name, _resource_name))
 
 
-def get_lock(name, resource_group_name=None):
+def get_lock(name, resource_group_name=None, resource_provider_namespace=None,
```

Comment:
This would also be the perfect time to fix #3476, as this is a significant change to this command signature.